### DISCUSSION TITLE
Difference between this implementation and the one present in SourceForge

### DISCUSSION THREAD

**Discussion by usmanmehmoodSSP** [fbarresi/Sharp7#40] [Q&A] [answered]
I had initially found Sharp7 on [SourceForge](https://snap7.sourceforge.net/) and I've been using it in one of my applications for half a year now. And it's been running reliably till now.

I just discovered this repo and I want to know what exactly is changed in this.
Is it the same? Or have there been any bugfixes or improvements?
How reliable is this to use?

I'd love to switch to this if this does not have any issues reliability wise (compared to the other one)

**Comment by fbarresi** [comment, marked as answer]
Hi!

The short version of the answer is: yes it is the same library as in sourceforge.

I forked the library from there years ago and notify the initial creator. Since then I made a nuget package from it and several unit test to ensure future changes will not change the behavior.

I applied some cosmetics to the code and applied also some Bugfixes. As far as I know the initial library get unmaintained years ago so no new changes were missed in this library. 

I think the initial idea of the project was to pack everything in a file and have no dependencies. So you can still downloading the single file in cs format. A better option without code duplication is a nuget package and this is it.

Hope I clarify your question. 

Best regards, 
FB

**Reply by usmanmehmoodSSP** [reply]
Hi,
Thank you for the reply. 
I will then use this in my projects instead of the original one.
